### PR TITLE
game-devices-udev-rules: init at 0.21

### LINKED ISSUE #231694: Package request: game-devices-udev
**Project description**

udev rules to make supported controllers available with user-grade permissions

**Metadata**

* homepage URL:
* source URL: https://codeberg.org/fabiscafe/game-devices-udev/
* license: mit
* platforms: linux

### PR DESCRIPTION
**Pull request by keenanweaver**
###### Description of changes

Adds 'game-devices-udev-rules' package. Resolves https://github.com/NixOS/nixpkgs/issues/231694. Also added me to the maintainers list.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Janik-Haag — APPROVED** [review]
LGTM; You might want to add a note in the description that people have to install it with `    services.udev.packages = [ pkgs.game-devices-udev-rules ];`or add a longDescription attribute describing this.

**Review by Janik-Haag — APPROVED** [review]

**Comment by keenanweaver** [comment]
Thanks for your time!

**Comment by Janik-Haag** [comment]
I will give this a few more days so other people get a chance to review and if no one else complains merge it.

File: pkgs/os-specific/linux/game-devices-udev-rules/default.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    install -D 71-nvidia-controllers.rules $out/lib/udev/rules.d/71-nvidia-controllers.rules
+    install -D 71-pdp-controllers.rules $out/lib/udev/rules.d/71-pdp-controllers.rules
+    install -D 71-personal_communication_systems_inc-controllers.rules $out/lib/udev/rules.d/71-personal_communication_systems_inc-controllers.rules
+    install -D 71-powera-controllers.rules $out/lib/udev/rules.d/71-powera-controllers.rules
+    install -D 71-razer-controllers.rules $out/lib/udev/rules.d/71-razer-controllers.rules
+    install -D 71-sony-controllers.rules $out/lib/udev/rules.d/71-sony-controllers.rules
+    install -D 71-sony-vr.rules $out/lib/udev/rules.d/71-sony-vr.rules
+    install -D 71-valve-controllers.rules $out/lib/udev/rules.d/71-valve-controllers.rules
+    install -D 71-valve-vr.rules $out/lib/udev/rules.d/71-valve-vr.rules
+    install -D 71-zeroplus_technology_corporation-controllers.rules $out/lib/udev/rules.d/71-zeroplus_technology_corporation-controllers.rules
```
Comment (NickCao) [inline pkgs/os-specific/linux/game-devices-udev-rules/default.nix:40]:
```suggestion
  install -Dm444 -t "$out/lib/udev/rules.d" *.rules
```

Comment (keenanweaver) [reply pkgs/os-specific/linux/game-devices-udev-rules/default.nix]: Yep, that's certainly **way** better than mine! The reason I separated them was for anyone who didn't want a particular rule they could overlay it out, but I think they could just create their own rule that precedes these ones instead.

File: pkgs/os-specific/linux/game-devices-udev-rules/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    domain = "codeberg.org";
+    owner = "fabiscafe";
+    repo = "game-devices-udev";
+    rev = version;
+    hash = "sha256-Yy91yDF5BSDTTlr/Pj8e0UklPooEdzvRW8mkhdHtHVo=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+  install -Dm444 -t "$out/lib/udev/rules.d" *.rules
```
Comment (NickCao) [inline pkgs/os-specific/linux/game-devices-udev-rules/default.nix:20]: The indentation is a bit off. Also please squash the commits.

**Review by Janik-Haag — APPROVED** [review]

File: pkgs/os-specific/linux/game-devices-udev-rules/default.nix (hunk 2 above)
Comment (keenanweaver) [reply pkgs/os-specific/linux/game-devices-udev-rules/default.nix]: Fixed. Thought I squashed them, but it's not showing. I'll see if I can figure that out.

**Review by NickCao — APPROVED** [review]

**Pull request merged**